zerver/openapi/zulip.yaml

```diff
@@ -6672,6 +6672,135 @@ paths:
                     description: |
                       An example JSON error response when attempting to deactivate the only
                       organization owner in an organization:
+  /users/me/alert_words:
+    get:
+      operationId: get-alert-words
+      summary: Get all alert words
+      tags: ["users"]
+      description: |
+        List all of the requesting user's configured [alert words][alert-words].
+
+        [alert words]: /help/pm-mention-alert-notifications#alert-words
+      responses:
+        "200":
+          description: Success.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/JsonSuccessBase"
+                  - $ref: "#/components/schemas/SuccessDescription"
+                  - additionalProperties: false
+                    properties:
+                      result: {}
+                      msg: {}
+                      alert_words:
+                        type: array
+                        description: |
+                          An array of strings where each string is an alert word specified by
+                          the requesting user.
+                        items:
+                          type: string
+                    example:
+                      {
+                        "result": "success",
+                        "msg": "",
+                        "alert_words": ["natural", "illustrious"],
+                      }
+    post:
+      operationId: add-alert-words
+      summary: Add alert words
+      tags: ["users"]
+      description: |
+        Add words to the requesting user's set of [alert words][alert-words].
+
+        [alert-words]: /help/pm-mention-alert-notifications#alert-words
+      parameters:
+        - name: alert_words
+          in: query
+          description: |
+            An array of strings to be added to the requesting user's set of alert words.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+              example: ["foo", "bar"]
+          required: false
```

Comment:
Is this an optional parameter? With curl in the dev environment I get this when I don't send the `alert_words` parameter ...

```console
{
   "code" : "REQUEST_VARIABLE_MISSING",
   "msg" : "Missing 'alert_words' argument",
   "result" : "error",
   "var_name" : "alert_words"
}

```

Reply:
I think I missed it while copy-pasting the schema. I think this should ideally be checked by the tests, whether parameters marked as `required: false` return an error in case they are not provided in the request.